ansible/roles/test/files/ptftests/radv_ipv6_ra_test.py

```diff
@@ -0,0 +1,240 @@
+
+    """
+    def create_icmpv6_router_advertisement_packet(self, dst_mac, dst_ip, src_mac, src_ip):
+        ether = scapy.layers.l2.Ether(dst=dst_mac, src=src_mac)
+        ip6 = scapy.layers.inet6.IPv6(src=src_ip, dst=dst_ip, fl=0, tc=0, hlim=255)
+        icmp6 = scapy.layers.inet6.ICMPv6ND_RA(code=0, M=1, O=0)
+        #NOTE: Test expects RA packet to contain route prefix as the first option
+        icmp6 /= scapy.layers.inet6.ICMPv6NDOptPrefixInfo(type=3, len=4)
+        rapkt = ether / ip6 / icmp6
+        return rapkt
+
+    """
+    @summary: Mask off fields we don't care about matching in RA packet
+    Match following fields of RA packet:-
+        1. Eth src and dest
+        2. Link local IPv6 src and dest
+        3. IP6 hop limit = 255
+        4. ICMPv6 RA type == 134 and code == 0
+        5. Flag M == 1 and O == 0
+        6. ICMPv6 prefix option (i.e type == 3 and len == 4)
+
+    """
+    def mask_off_dont_care_ra_packet_fields(self, rapkt):
+        masked_rapkt = Mask(rapkt)
+        masked_rapkt.set_ignore_extra_bytes()
+
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.IPv6, "version")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.IPv6, "tc")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.IPv6, "fl")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.IPv6, "plen")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.IPv6, "nh")
+
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "cksum")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "chlim")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "H")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "prf")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "P")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "res")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "routerlifetime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "reachabletime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "retranstimer")
+
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "prefixlen")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "L")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "A")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "R")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "res1")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "validlifetime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo,
+                                    "preferredlifetime")
```

Comment:
Suggest aligning this with the first parameter in the line above (i.e., add 7 spaces).